src/sql/workbench/services/query/common/queryManagement.ts

```diff
@@ -266,6 +271,23 @@ export class QueryManagementService implements IQueryManagementService {
 		});
 	}
 
+	public changeConnectionUri(newUri: string, oldUri: string): Promise<void> {
+		let item = this._queryRunners.get(oldUri);
+		if (!item) {
+			this._logService.error(`No query runner found for old URI : '${oldUri}'`);
+			throw new Error(nls.localize('queryManagement.noQueryRunnerForUri', 'Could not find Query Runner for uri: {0}', oldUri));
+		}
+		if (this._queryRunners.get(newUri)) {
```

Comment:
Same here about overwriting an existing connection

Reply:
It shouldn't reach this point as it would be caught in changeConnectionUri, but in the unlikely case it does, I'll add a catch statement to change the uri back to the old one.

Reply:
See my comment for QueryModelService.ts on how an error in each step will work.